mempool/mempool_test.go

```diff
@@ -166,17 +169,20 @@ func TestRestoreMempool(t *testing.T) {
 	lenDB, err = poolRestored.LenDB()
 	require.NoError(t, err)
 	require.Equal(t, 3, lenDB)
-	require.Equal(t, 3, poolRestored.Len())
+	lenRestored := poolRestored.Len()
+	require.Equal(t, 3, lenRestored)
```

Comment:
I believe this equality might be redundant. You are already checking the length with `transactions` below which is created with size three. You could make it a static array if you want to guarantee it will never grow.